tests/smoke.spec.ts

```diff
@@ -1391,3 +1392,105 @@ test.describe("OSDEV-1812: Smoke: Moderation queue page is can be opened through
+    await expect(page.getByRole("button", { name: "REGISTER" })).toBeVisible();
+    await expect(page.getByRole("button", { name: "LOG IN" })).toBeVisible();
+  })
+
+  test("An authorized user can download a list of facilities with amounts 7000 - 9900 in xlsx.", async ({
+    page,
+  }) => {
+    // Log in to the main page
+    const { BASE_URL } = process.env;
+    await page.goto(`${BASE_URL}/facilities/?countries=AO&countries=BE&countries=PL&sort_by=contributors_desc`!);
+    page.getByRole("button", { name: "Download" }).click({ force: true });
+
+    // Check that the menu item is visible
+    const menuItem = page.getByRole("menuitem", { name: "Excel" });
+    await expect(menuItem).toBeVisible();
+    menuItem.click({ force: true });
+
+    // Log in to the main page
+    page.getByRole("button", { name: "LOG IN" }).click();
+    const { USER_EMAIL, USER_PASSWORD } = process.env;
+    await page.getByLabel("Email").fill(USER_EMAIL!);
+    await page.getByRole("textbox", { name: "Password" }).fill(USER_PASSWORD!);
+    await page.getByRole("button", { name: "Log In" }).click();
+    page.getByRole("button", { name: "Download" }).click({ force: true });
+    page.getByRole("menuitem", { name: "Excel" }).click({ force: true });
+
+    // Download the file
+    const downloadPath = path.resolve(__dirname, "downloads");
+    const downloadPromise = page.waitForEvent("download");
+    const download = await downloadPromise;
+    const filePath = path.join(downloadPath, download.suggestedFilename());
+    await download.saveAs(filePath);
+    const fileExists = fs.existsSync(filePath);
+    const fileName = path.basename(filePath);
+
+    expect(fileExists).toBe(true);
+    expect(fileName).toContain("facilities");
+    expect(fileName).toContain(".xlsx");
+
+    // Get the first sheet
+    async function readXlsx(filePath: string) {
+      const fileName = path.basename(filePath);
+      if (!fileName.endsWith(".xlsx")) throw new Error("Invalid file type");
+      const workbook = new ExcelJS.Workbook();
+      await workbook.xlsx.readFile(filePath);
+      const rows: CellValue[][] = [];
+      if (workbook.worksheets.length === 0) {
+        throw new Error("Workbook contains no sheets");
+      }
+      const worksheet = workbook.worksheets[0];
+      if (!worksheet) {
+        throw new Error("First worksheet is null");
+      }
+      worksheet.eachRow((row: Row) => {
+        if (row.values && Array.isArray(row.values)) {
+          rows.push(row.values.slice(1));
+        }
+      });
+      return rows;
+    }
+    // Check that the number of facilities is visible
+    const results = page.getByText(/^\d+ results$/);
+    await expect(results).toBeVisible();
+
+    // Get count of Facilities output on the  UI
+    const text = await results.textContent();
+    const numberOfFacilities = parseInt(text?.match(/\d+/)?.[0] || "0", 10);
+    const headerRow = 1;
```

Comment:
This variable seems redundant, it does not abstract any logic and we don't re-use it's value I would suggest we just do `+ 1` in the logic and remove it.

Reply:
Can you please explain what you mean by `magic numbers`. Not sure I fully understand what you are trying to avoid.